src/module_library/maintenance_respiration.h

```diff
@@ -0,0 +1,128 @@
+#ifndef MAINTENANCE_RESPIRATION_H
+#define MAINTENANCE_RESPIRATION_H 
+
+#include "../framework/module.h"
+#include "../framework/state_map.h"
+#include "BioCro.h"  // for temprature_response 
+
+namespace standardBML
+{
+/**
+ * @class maintenance_respiration
+ *
+ * @brief Calculates the change in plant organ biomasses due to maintenance respiration.
+ *
+ * The amount that each plant component respires is determined as a
+ * percentage of its current biomass. 
+ * This ideas is from this paper: 
+ * (https://doi.org/10.1016/j.fcr.2010.07.007)
+ */
+class maintenance_respiration : public differential_module
+{
+   public:
+    maintenance_respiration(
+        state_map const& input_quantities,
+        state_map* output_quantities)
+        : differential_module{},
+
+          // Get references to input quantities
+          Leaf{get_input(input_quantities, "Leaf")},
+          Stem{get_input(input_quantities, "Stem")},
+          Root{get_input(input_quantities, "Root")},
+          Rhizome{get_input(input_quantities, "Rhizome")},
+          Grain{get_input(input_quantities, "Grain")},
+          Shell{get_input(input_quantities, "Shell")},
+          temp{get_input(input_quantities, "temp")},
+          mrc_leaf{get_input(input_quantities, "mrc_leaf")},
+          mrc_stem{get_input(input_quantities, "mrc_stem")},
+          mrc_root{get_input(input_quantities, "mrc_root")},
+          mrc_grain{get_input(input_quantities, "mrc_grain")},
+
+          // Get pointers to output quantities
+          Leaf_op{get_op(output_quantities, "Leaf")},
+          Stem_op{get_op(output_quantities, "Stem")},
+          Root_op{get_op(output_quantities, "Root")},
+          Rhizome_op{get_op(output_quantities, "Rhizome")},
+          Grain_op{get_op(output_quantities, "Grain")},
+          Shell_op{get_op(output_quantities, "Shell")}
+    {
+    }
+    static string_vector get_inputs();
+    static string_vector get_outputs();
+    static std::string get_name() { return "maintenance_respiration"; }
+
+   private:
+    // References to input quantities
+    const double& Leaf;
+    const double& Stem;
+    const double& Root;
+    const double& Rhizome;
+    const double& Grain;
+    const double& Shell;
+    const double& temp;
+    const double& mrc_leaf;
+    const double& mrc_stem;
+    const double& mrc_root;
+    const double& mrc_grain;
+
+    // Pointers to output quantities
+    double* Leaf_op;
+    double* Stem_op;
+    double* Root_op;
+    double* Rhizome_op;
+    double* Grain_op;
+    double* Shell_op;
+
+    // Implement the pure virtual function do_operation():
+    void do_operation() const override final;
+};
+
+string_vector maintenance_respiration::get_inputs()
```

Comment:
The `get_inputs()` function is missing some of the module inputs: `"mrc_leaf"`, `"mrc_stem"`, `"mrc_root"`, and `"mrc_grain"`.

Because of this, the module is not fully functional. For example, typing `module_info('BioCro:maintenance_respiration')` produces this error:

```
Error: could not create the module
Additional details:
Thrown by get_input: the quantity 'mrc_leaf' was not defined in the state_map.
```

I found this issue by running the package tests. In general, it is a good idea to make sure all tests pass before submitting a PR. There are several ways to run the tests. I usually do it from R via `test_dir('biocro/tests/testthat')` (since I am usually working one directory above the `biocro` source directory). Other options are described here: https://biocro.github.io/BioCro-documentation/latest/bookdown/running-the-testthat-tests.html

Reply:
Thanks for bring this up! It slipped my mind. At some point, the testing was in my routine of making changes.